desktop/ui/src/main/java/org/datacleaner/documentation/ComponentDocumentationBuilder.java

```diff
@@ -0,0 +1,181 @@
+        _breadcrumbs = breadcrumbs;
+        _freemarkerConfiguration = new Configuration(Configuration.DEFAULT_INCOMPATIBLE_IMPROVEMENTS);
+        _freemarkerConfiguration.setClassForTemplateLoading(this.getClass(), ".");
+        try {
+            _template = _freemarkerConfiguration.getTemplate("template.html");
+        } catch (Exception e) {
+            throw new IllegalStateException("Failed to load template", e);
+        }
+    }
+
+    /**
+     * Creates the reference documentation for a {@link Component}.
+     * 
+     * @param componentDescriptor
+     *            the {@link ComponentDescriptor} of the {@link Component} of
+     *            interest.
+     * @param outputStream
+     *            the target {@link OutputStream} to write to
+     * @throws IOException
+     */
+    public void write(ComponentDescriptor<?> componentDescriptor, OutputStream outputStream)
+            throws IOException {
+
+        final Map<String, Object> data = new HashMap<>();
+
+        try {
+            data.put("breadcrumbs", _breadcrumbs);
+            data.put("component", new ComponentDocumentationWrapper(componentDescriptor));
+
+            {
+                final Set<ConfiguredPropertyDescriptor> configuredProperties = componentDescriptor
+                        .getConfiguredProperties();
+                final List<ConfiguredPropertyDescriptor> properties = new ArrayList<ConfiguredPropertyDescriptor>(
+                        configuredProperties);
+                final List<ConfiguredPropertyDocumentationWrapper> propertyList = new ArrayList<>();
+                for (ConfiguredPropertyDescriptor property : properties) {
+                    ConfiguredPropertyDocumentationWrapper wrapper = new ConfiguredPropertyDocumentationWrapper(
+                            property);
+                    propertyList.add(wrapper);
+                }
+
+                data.put("properties", propertyList);
+            }
+
+            { // Attach the image
+                final Image descriptorIcon = IconUtils.getDescriptorIcon(componentDescriptor).getImage();
+
+                /* We need a buffered image type in order to obtain the */
+                final BufferedImage bufferedImage = toBufferedImage(descriptorIcon);
+                final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                ImageIO.write(bufferedImage, "png", baos);
+                byte[] imageInByte = baos.toByteArray();
+                baos.flush();
+                baos.close();
+
+                /* Encode the image */
+                final String encodedImage = Base64.encodeBase64String(imageInByte);
+
+                /*
+                 * Atach the prefix that will make html <img> know how to decode
+                 * the image
+                 */
+                final String iconHtmlRepresentation = HTMLBASE64_PREFIX + encodedImage;
+
+                data.put("icon", iconHtmlRepresentation);
```

Comment:
Very much nitpicking, but it would be nice to keep the icon generation separate in its own class.